### PR TITLE
[WIP] increased code coverage

### PR DESCRIPTION
**Pull request by iamrajiv**
#### References to other Issues or PRs or Relevant literature

#19

#### Brief description of what is fixed or changed
improved tests in order to increase PyDataStructs code coverage

### CODE REVIEW

**Comment by czgdp1807** [comment]
Are you participating through GSSoC, 2020?

**Comment by iamrajiv** [comment]
@czgdp1807 Yes

File: pydatastructs/graphs/__init__.py [hunk 1]
```diff
@@ -11,3 +11,9 @@
     AdjacencyList
 )
 __all__.extend(adjacency_list.__all__)
+
+from . import adjacency_matrix
+from .adjacency_matrix import (
+    AdjacencyMatrix
+)
+__all__.extend(adjacency_matrix.__all__)
```
Comment (czgdp1807) [inline pydatastructs/graphs/__init__.py]: IMO, `AdjacencyMatrix` and `AdjacencyList` shouldn't be there in the global namespace. Please remove them from here and make suitable changes to the import statements.

Comment (iamrajiv) [reply pydatastructs/graphs/__init__.py]: @czgdp1807 but `AdjacencyList` was there from starting should I remove that also

Comment (czgdp1807) [reply pydatastructs/graphs/__init__.py]: Yes.

File: pydatastructs/trees/heaps.py [hunk 2]
```diff
@@ -77,7 +77,7 @@ def __new__(cls, elements=None, heap_property="min"):
         elif heap_property == "max":
             obj._comp = lambda key_parent, key_child: key_parent >= key_child
         else:
-            raise ValueError("%s is invalid heap property"%(heap_property))
```
Comment (czgdp1807) [inline pydatastructs/trees/heaps.py]: Please restore. This is more informative.

**Comment by czgdp1807** [comment]
If you observe any differences in the observed output and the expected output then please open a separate issue.

File: .coveragerc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[report]
+exclude_lines =
+    pragma: no cover
+
+    raise NotImplementedError
+    raise ValueError
```
Comment (czgdp1807) [inline .coveragerc]: IMO, this should be removed. `ValueError` covers the corner cases.
Is it possible to include something like,
```
exclude_files =
       pragma: no cover

       *test*
```

Comment (iamrajiv) [reply .coveragerc]: @czgdp1807 I have removed the line `raise ValueError` and  tested locally and it works fine and does not affect the coverage
```
exclude_files =
       pragma: no cover

       *test*
```
above code is not working its giving invalid syntax

Comment (iamrajiv) [reply .coveragerc]: @czgdp1807 thanks for guidance I have removed the line `raise ValueError` and tested locally and it works fine
```
[run]
parallel = True
source = pydatastructs
omit =
    */tests/*
```
I have added the above code to exclude the tests in `.coveragerc` and merged `coveragerc_travis` in `.coveragerc`

**Comment by czgdp1807** [comment]
Delete the system generated files.

File: .gitignore [hunk 4]
```diff
@@ -1,12 +1,55 @@
...
+
+# Backup Files #
+################
+*.bak
+*.swp
+
+# Editor Files #
+################
 .vscode/
-.pytest_cache/
```
Comment (czgdp1807) [inline .gitignore]: Add this under python files.

File: .gitignore [hunk 5]
```diff
@@ -1,12 +1,55 @@
...
-pre_commit.ps1
```
Comment (czgdp1807) [inline .gitignore]: Under OS generated files.

File: .gitignore [hunk 6]
```diff
@@ -1,12 +1,55 @@
...
-htmlcov
-*coverage*
-*.egg-info/
```
Comment (czgdp1807) [inline .gitignore]: Under OS generated files.

**Comment by czgdp1807** [comment]
Please delete the `DS_Store` binary file.

**Comment by czgdp1807** [comment]
Looks ready to go.

**Comment by manaswinidas** [comment]
> Looks ready to go. ping @codezonediitj/gssoc2020 Any comments?

LGTM too.
